nix_update/update.py

```diff
@@ -270,6 +270,104 @@ def update_cargo_lock(
                 print(line, end="")
 
 
+def update_pubspec_lock(opts: Options, filename: str) -> None:
+    with tempfile.TemporaryDirectory() as tempdir:
+        with Path(filename).open() as f:
+            match = re.search(
+                r"pubspecLock\s*=\s*lib\.importJSON\s+([^;]+)\s*;",
+                f.read(),
+            )
+            if match is None:
+                return
+            dst = Path(match.group(1))
+            if not dst.is_absolute():
+                dst = Path(filename).resolve().parent / dst
+                dst = dst.resolve()
+        if opts.generate_lockfile:
+            generate_lockfile(opts, filename, "pubspec")
+            with Path.open(
+                Path(tempdir) / "pubspec.lock.json",
+                "w",
+                encoding="utf-8",
+            ) as f:
+                run(
+                    [
+                        "nix",
+                        "run",
+                        "-L",
+                        "--impure",
+                        *opts.extra_flags,
+                        "github:NixOS/nixpkgs/nixos-unstable#yq",
```

Comment:
why do we do this here?

Reply:
I am bit worried about breaking stuff here. The yaml standard is not very well defined and very complex.